database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowStoreEntryComponent.scala

```diff
@@ -75,11 +75,11 @@ trait WorkflowStoreEntryComponent {
   /**
     * Returns up to "limit" startable workflows, sorted by submission time.
     */
-  val fetchStartableWorkflows = Compiled(
-    (limit: ConstColumn[Long],
-     heartbeatTimestampTimedOut: ConstColumn[Timestamp],
-     excludeWorkflowState: Rep[String]
-    ) => {
+  def fetchStartableWorkflows(limit: Long,
+     heartbeatTimestampTimedOut: Timestamp,
+     excludeWorkflowState: String,
+     excludedGroups: Set[String]
+    ): Query[WorkflowStoreEntries, WorkflowStoreEntry, Seq] = {
```

Comment:
I need to verify that removing this query from `Compiled` status doesn't introduce unacceptable performance degradation.

It is a known issue in Slick [0] that `inSet` and `Compiled` do not get along.

[0] https://github.com/slick/slick/issues/718

Reply:
What form will this verification take?

Reply:
What form will this verification take?

Reply:
My plan is to add a case to the suite that calls `workflowStore.fetchStartableWorkflows` as quickly as possible in a loop for a fixed number of iterations and measure its execution time. I think we have microbenchmarks like this elsewhere in the codebase. Didn't get to it today due to various UL fires.

The target is somewhat lenient, the query is executed at most once per second per instance: https://github.com/broadinstitute/firecloud-develop/blob/dev/base-configs/cromwell/cromwell.conf.ctmpl#L133-L139

Reply:
Alternatively I would love to find a solution that keeps `Compiled`, I bashed my head against it unsuccessfully for a good while last week.

Reply:
Wrote a speed measurement test case [0] and ran it  this branch and `develop`.

Time is seconds to execute the query 500 times.

Non-compiled:
5.9, 5.6, 5.5
Avg. 5.7 seconds

Compiled:
4.0, 3.9, 4.1
Avg. 4.0 seconds

So there is a noticeable difference, but it's not one that I'm worried about when considering the current configured rate of 1 query per second.

[0]

```
it should "be acceptably fast" taggedAs DbmsTest in {
  (for {
    submissionResponsesExcluded <- workflowStore.add(sourceFilesCollection)
    _ = (for (i <- 1 to 500) {
      Await.result(workflowStore.fetchStartableWorkflows(10, "A08", 1.second), 1.second)
    })
  } yield ()).futureValue
}
```